transformer_engine/common/include/transformer_engine/transformer_engine.h

```diff
@@ -393,6 +393,105 @@ int nvte_is_non_tn_fp8_gemm_supported();
 */
 void nvte_memset(void *ptr, int value, size_t size_in_bytes, cudaStream_t stream);
 
+/*! \brief TE Grouped Tensor type
+ *
+ * NVTEGroupedTensor is a collection of tensors with potentially different shapes
+ * but the same dtype and scaling mode. It does not own the memory it points to.
+ */
+typedef void *NVTEGroupedTensor;
+
+/*! \enum NVTEGroupedTensorParam
+ *  \brief Indicates the kind of the grouped tensor parameter to set/get.
+ */
+enum NVTEGroupedTensorParam {
+  kNVTEGroupedRowwiseData = 0,        /*!< Data usable in rowwise manner */
+  kNVTEGroupedColumnwiseData = 1,     /*!< Data usable in columnwise manner */
+  kNVTEGroupedScale = 2,              /*!< Scale tensor */
+  kNVTEGroupedAmax = 3,               /*!< Amax tensor */
+  kNVTEGroupedRowwiseScaleInv = 4,    /*!< Scale inverse tensor for decoding Rowwise Data */
+  kNVTEGroupedColumnwiseScaleInv = 5, /*!< Scale inverse tensor for decoding Columnwise Data */
+  kNVTEGroupedColumnwiseAmax = 6,     /*!< Columnwise Amax tensor */
+  kNVTEGroupedFirstDims = 7,          /*!< First dimension sizes (device pointer to size_t array) */
+  kNVTEGroupedSecondDims = 8,      /*!< Second dimension sizes (device pointer to size_t array) */
+  kNVTEGroupedCumulativeSizes = 9, /*!< Cumulative tensor sizes (device pointer to size_t array) */
+  kNVTENumGroupedTensorParams
+};
+
```

Comment:
I would prefer to avoid completely forking off the API for the grouped tensor from the regular tensor - most of those parameters are the same between the two and have conceptually the same meaning. Also the comment is confusing - we do not want the user to provide a literal size_t* value so we should not tell them so.